modules/runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/model/impl/RegistrationInfoImpl.java

```diff
@@ -342,8 +343,8 @@ protected ComponentInstance createComponentInstance() {
             String msg = "Failed to instantiate component: " + implementation;
             log.error(msg, e);
             msg += " (" + e.toString() + ')';
-            Framework.getRuntime().getMessageHandler().addError(msg);
-            throw e;
+            Framework.getRuntime().getMessageHandler().addMessage(Level.ERROR, msg);
+            return null;
```

Comment:
I don't like `null`,  they are often error prone. What do you think about moving the content of this method in the `activate` method?
As it we drop the null thing + we have the full error management (with the early return) at the same place.
Wdyt?

Reply:
Ok but it so, it appears more consistent to catch an error at component activation too...
That will change the behaviour described in the RuntimeInitializationTestComponent behaviour (see "error at activation will bubble up and will not be caught by message handler" in my test comments).

We would have in the final RegistrationInfo#activate method the following process:
- try/catch on component instantiation (add error to runtime if so and return)
- try/catch on activation (new behaviour: add error to runtime if so and return instead of bubbling up the error)
- try/catch on each of the extensions contributions
- try/catch on each of the pending contributions

WDYT?

Reply:
I will include the change so that you can see what it looks like, i can revert it if we're not happy with it or there are unexpected impacts.

Reply:
(pushed)

Reply:
I like it 👍